# Issue: Fix Compiler Error CS1061 and Shader Include Errors

## Description
The project currently fails to compile and has broken shaders due to missing definitions and incorrect include paths.

## Errors

### 1. Compiler Error
**File:** `Assets\Scripts\DecorateRoomScreen.cs`
**Error:**
```
Assets\Scripts\DecorateRoomScreen.cs(487,28): error CS1061: 'WellDoneScreen.InitArguments' does not contain a definition for 'onReplay'
```
**Cause:**
The `WellDoneScreen.InitArguments` struct definition in `WellDoneScreen.cs` is missing the `onReplay` field, but it is being referenced in `DecorateRoomScreen.cs`.

### 2. Shader Errors
**Files:**
*   `Assets\TextMesh Pro\Shaders\TMP_SDF SSD.shader`
*   `Assets\TextMesh Pro\Shaders\TMP_SDF-Mobile SSD.shader`

**Error:**
```
Shader error in 'TextMeshPro/Distance Field SSD': Couldn't open include file 'TMPro_Properties.cginc'
```

**Cause:**
The shaders attempt to include `TMPro_Properties.cginc` and `TMPro.cginc` from the same directory, but these files are located in `Assets/TextMesh Pro/Resources/Shaders/`.

## Proposed Fixes

### Fix CS1061 in `WellDoneScreen.cs`
1.  Add `public Action onReplay;` to the `InitArguments` struct.
2.  Pass the `onReplay` action to the `WellDoneContainer` in the `DoAnimation` method.

### Fix Shader Includes
Update the `#include` paths in the affected shaders to use the correct relative path:
```glsl
#include "../Resources/Shaders/TMPro_Properties.cginc"
```

--- a/Assets/Scripts/WellDoneScreen.cs
+++ b/Assets/Scripts/WellDoneScreen.cs
@@ -57,6 +57,8 @@ public class WellDoneScreen : MonoBehaviour
 		public string mainText;
 
 		public Action onComplete;
+
+		public Action onReplay;
 	}
 
 	private sealed class _003C_003Ec__DisplayClass7_0
@@ -101,6 +103,7 @@ public class WellDoneScreen : MonoBehaviour
 				WellDoneContainer.InitArguments initArguments = default(WellDoneContainer.InitArguments);
 				this._003C_003E8__1.complete = false;
 				initArguments.onComplete = new Action(this._003C_003E8__1._003CDoAnimation_003Eb__0);
+				initArguments.onReplay = wellDoneScreen.initArguments.onReplay;
 				wellDoneScreen.container.Show(initArguments);
 			}
 			if (this._003C_003E8__1.complete)

--- a/Assets/TextMesh Pro/Shaders/TMP_SDF SSD.shader
+++ b/Assets/TextMesh Pro/Shaders/TMP_SDF SSD.shader
@@ -124,8 +124,8 @@ SubShader {
 
         #include "UnityCG.cginc"
         #include "UnityUI.cginc"
-        #include "TMPro_Properties.cginc"
-        #include "TMPro.cginc"
+        #include "../Resources/Shaders/TMPro_Properties.cginc"
+        #include "../Resources/Shaders/TMPro.cginc"
 
         struct vertex_t {
             UNITY_VERTEX_INPUT_INSTANCE_ID

--- a/Assets/TextMesh Pro/Shaders/TMP_SDF-Mobile SSD.shader
+++ b/Assets/TextMesh Pro/Shaders/TMP_SDF-Mobile SSD.shader
@@ -94,7 +94,7 @@ SubShader {
 
 		#include "UnityCG.cginc"
 		#include "UnityUI.cginc"
-		#include "TMPro_Properties.cginc"
+		#include "../Resources/Shaders/TMPro_Properties.cginc"
 
 		#include "TMPro_Mobile.cginc"
 

